restli-common/src/main/java/com/linkedin/restli/common/validation/RestLiDataValidator.java

```diff
@@ -436,6 +437,19 @@ public ValidationResult validateOutput(RecordTemplate dataTemplate, MaskTree pro
    * @throws IllegalArgumentException if any argument is null or if the provided data template has no data
    */
   protected ValidationResult validateOutputAgainstSchema(RecordTemplate dataTemplate, DataSchema validatingSchema)
+  {
+    if (validatingSchema == null)
+    {
+      throw new IllegalArgumentException("Validating schema is null");
+    }
+
+    return validateOutputAgainstSchema(dataTemplate,
+        () -> ValidateDataAgainstSchema.validate(dataTemplate.data(), validatingSchema, new ValidationOptions(),
+            new DataSchemaAnnotationValidator(validatingSchema)));
+  }
+
+  protected ValidationResult validateOutputAgainstSchema(RecordTemplate dataTemplate,
+      Supplier<ValidationResult> validationResultSupplier)
```

Comment:
In regards to `protected Validator getValidatorForOutput(DataSchema validatingSchema);`, we would be overriding this in `RestLiDataSchemaDataValidator` right?

In this case, the overridden method would look something like 
```protected Validator getValidatorForOutput(DataSchema validatingSchema) { return _schemaValidator; }```

It felt a little strange to me to do something like this, since we are ignoring the input validatingSchema parameter.